src/compiler/checker.ts

```diff
@@ -3216,9 +3216,7 @@ namespace ts {
         }
 
         function checkAndReportErrorForUsingTsPlusTypeAsValue(errorLocation: Node, name: __String, meaning: SymbolFlags, globalImport: TsPlusGlobalImport): boolean {
-            if (meaning & (SymbolFlags.Value & ~SymbolFlags.NamespaceModule) &&
-                meaning !== SymbolFlags.Type &&
-                meaning !== (SymbolFlags.Type | SymbolFlags.Namespace)) {
+            if ((meaning & SymbolFlags.Value) === SymbolFlags.Value) {

```

Comment:
For future reference: I'm pretty but this works, but if inexplicable/incorrect errors saying `'x' only refers to a type, but is being used as a value here` crop up from using global imports, it's probably this flag check that needs to be modified. Bit-flags are strange.

Reply:
indeed bitmask based checks are always hard

Reply:
Doing `A & Mask === Mask` is what I used to do in the past, it always felt strange to me that TS is checking usually like `if (X & Mask) {}` and don't fully understand it